labgrid/driver/usbstoragedriver.py

```diff
@@ -47,6 +54,69 @@ def on_activate(self):
     def on_deactivate(self):
         pass
 
+    def _wait_for_medium(self, partition=None):
+        timeout = Timeout(self.WAIT_FOR_MEDIUM_TIMEOUT)
+        while not timeout.expired:
+            if self.get_size(partition) > 0:
+                break
+            time.sleep(self.WAIT_FOR_MEDIUM_SLEEP)
+        else:
+            raise ExecutionError("Timeout while waiting for medium")
+
+    def _pmount(self, partition):
+        mount_args = ["pmount", self.storage.path + str(partition)]
+        self._wait_for_medium(partition)
+        processwrapper.check_output(self.storage.command_prefix + mount_args)
+
+    def _pumount(self, partition):
+        dev_path = self.storage.path + str(partition)
+        for i in range(self.PUMOUNT_MAX_RETRIES):
+            sync_args = ["sync", dev_path]
+            processwrapper.check_output(self.storage.command_prefix + sync_args)
+
+            umount_args = ["pumount", dev_path]
+            try:
+                processwrapper.check_output(self.storage.command_prefix + umount_args)
+            except subprocess.CalledProcessError as e:
+                if e.returncode == 5:
+                    self.logger.info('umount: %s: target is busy; wait for %s s',
+                                     dev_path, self.PUMOUNT_BUSY_WAIT)
+                    time.sleep(self.PUMOUNT_BUSY_WAIT)
+                    continue
+                raise e
+            break
+
+    @Driver.check_active
+    @step(args=['filenames', 'target_dir', 'partition'])
+    def copy_files(self, filenames, target_dir, partition=1):
+        """
+        Copies the file(s) specified by filename(s) to the
+        bound USB storage partition.
+
+        Args:
+            filenames (List[str]): path(s) to the file(s) to be copied to the bound USB storage
+                partition.
+            target_dir (str): optional, target directory to copy to (defaults to partition root
+                directory)
```

Comment:
This parameter seems not to be optional, but I guess using "/" as a default might make sense?

Reply:
Jep, definitely.

Reply:
Fixed